src/components/map/MapService.js

```diff
@@ -828,13 +804,48 @@ goog.require('ga_urlutils_service');
           }
         };
 
+        // Load layers config
+        var lastLangUsed;
+        var loadLayersConfig = function(lang) {
+          if (lastLangUsed == lang) {
+            return;
+          }
+          lastLangUsed = lang;
+          var url = getLayersConfigUrl(lang);
+          return $http.get(url).then(function(response) {
+            if (!layers) { // First load
+              layers = response.data;
+              // We register events only when layers are loaded
+              $rootScope.$on('$translateChangeEnd', function(event, newLang) {
+                loadLayersConfig(newLang.language);
+              });
+
+              $rootScope.$on('gaTimeSelectorChange', function(event, time) {
+                currentTime = time;
+              });
+
+            } else { // Only translations has changed
+              layers = response.data;
+              $rootScope.$broadcast('gaLayersTranslationChange', layers);
+            }
+          });
+        };
+
+        // Load layers configuration with value from permalink
+        // gaLang.get() never returns an undefined value on page load.
+        var configP = loadLayersConfig(gaLang.get());
+
+        /**
+         * Get the promise of the layers config requets
+         */
+        this.loadConfig = function() {
+          return configP;
+        };
+
         /**
          * Reurn an array of pre-selected bodId from current topic
          */
         this.getSelectedLayers = function() {
-          if (!layers || !gaTopic.get()) {
-            return;
-          }
           return gaTopic.get().selectedLayers;
```

Comment:
I think we should remove this function here, as (pre-) selected layers depend on topic. It should not be part of gaLayers.

In any case, the gaTopic.loadConfig() promise must be resolved before people can call this...this could be indicated in a comment.